idaes/core/unit_model.py

```diff
@@ -656,5 +657,13 @@ def initialize(blk, state_args=None, outlvl=idaeslog.NOTSET,
         # Release Inlet state
         blk.control_volume.release_state(flags, outlvl)
 
+        if (results is not None and (
+                results.solver.termination_condition !=
+                TerminationCondition.optimal or
+                results.solver.status != SolverStatus.ok)):
```

Comment:
Instead of repeating this everywhere, why not just:
```python
from pyomo.environ import check_optimal_termination
# ...
if not check_optimal_termination(results):
   raise InitializationError(#...
```

Reply:
Mostly because I copied code that I already had that did it this way.